src/controllers/RootController.ts

```diff
@@ -23,13 +20,18 @@ export interface GeneralPrivateInfo {
 
 export interface GeneralPublicInfo {
   loginMethod: LoginMethods;
+  setupDone: boolean,
 }
 
 @Route('')
 export class RootController extends Controller {
   @Post('setup')
-  public async postSetup(@Body() params: SetupParams): Promise<void> {
-    return new ServerSettingsService().initialSetup(params);
+  public async postSetup(@Body() params: SetupParams, @Request() req: express.Request): Promise<void> {
+    const user = await new ServerSettingsService().initialSetup(params);
```

Comment:
Please also validate the input. Rule #1 of webdevelopment is to never trust your frontend. I can fairly easy remove any form constraints there and for example send a password that does not meet the requirements. If the server does not check this, I have created a weak password. Therefore, the server should also check the same thing the frontend checks. Note that TSOA already checks for basic stuff like object structure and type. We only need to check whether the values in the input fields actually make sense. In this case, it is probably only the password constraints and whether some fields do not contain an empty string.